### PR TITLE
Only answer if your a thief once per location

### PR DESCRIPTION
**Pull request by alkemann**
#### Summary
```SUMMARY: Interface "Only answer if you are a thief once per location grab"```

#### Purpose of change
When stealing a lot of items, you have to keep answering yes on each item. With this change the pickup loop remembers your choice for the entire location. If you say "NO I am not thief", the player only picks up all the items that is not tagged as owned.

#### Describe the solution
This can have 3 values; `ask`, `steal` or `honest` and starts out being `ask`. Upon encountering an action that would be stealing, the player is queried like before, but the answer is stored. It is reset after all pickup is done. There is also a new key binding that cycles between the 3 values.

#### Describe alternatives you've considered
I considered using enums, but the `Creature.set_value` only allows storing of strings, so then I considered constants, but ended up with just hardcoded strings.

They keybinding part can be left out and this PR is still a nice improvement.

### CODE REVIEW

File: src/pickup.cpp [hunk 1]
```diff
@@ -191,8 +191,15 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
     const auto wield_check = u.can_wield( newit );
     if( newit.has_owner() &&
         newit.get_owner() != g->faction_manager_ptr->get( faction_id( "your_followers" ) ) ) {
-        if( !query_yn( "Picking up this item will be considered stealing, continue?" ) ) {
-            return false;
+        if( u.thief == THIEF_UNSET ) { // Has the player given input on if stealing is ok?
+            if( query_yn( "Picking up this item will be considered stealing, continue?" ) ) {
```
Comment (BevapDin) [inline src/pickup.cpp]: Needs to be translatable - see "doc/TRANSLATING.md".

File: src/player.cpp [hunk 2]
```diff
@@ -514,6 +514,8 @@ player::player() :
     nv_cached = false;
     volume = 0;
 
+    thief = THIEF_UNSET; // Is player willing to steal
```
Comment (BevapDin) [inline src/player.cpp]: Nearly nobody is going to look into the default constructor for documentation of a member. Put documentation where you declare that member (in the header file).

File: src/pickup.cpp (hunk 1 above)
Comment (alkemann) [reply src/pickup.cpp]: Notice that I didnt add that question, but I dont mind adding _() around it. :)

Comment (davidpwbrown) [reply src/pickup.cpp]: Yep, that one is my bad.

File: src/player.h [hunk 3]
```diff
@@ -185,13 +191,23 @@ static const std::array< std::string, PMM_COUNT > player_movemode_str = { {
...
         player &operator=( player && );
 
+        /**
+         * Toggle attitude towards stolen goods, ask if UNSET
+         *
+         * UNSET: Ask player when attempting to steal
+         * HONEST: Do not pick up stolen items, even if selected in pickup
+         * STEAL: Pick up stolen items without query
+         */
+        thief_mode thief;
```
Comment (kevingranade) [inline src/player.h]: Please use creature::values for musc state like this.
https://github.com/CleverRaven/Cataclysm-DDA/blob/9119b820d5503409c180c5e5bc71db4744ebb6c9/src/creature.h#L351

Comment (alkemann) [reply src/player.h]: Ah, thanks I will try that

**Comment by alkemann** [comment]
Updated the description to reflect that I have moved from player prop to the Creature.set_value. Also I have added a keybinding if a player does not want to be asked once per location either.

File: src/pickup.cpp [hunk 4]
```diff
@@ -191,8 +192,39 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
...
+            "THIEF_ASK" ) { // Has the player given input on if stealing is ok?
+            // y/n query_yn( _( "Picking up this item will be considered stealing, continue?" ) )
+            std::string answer = query_popup()
+                .context( "YESNOALWAYSNEVER" )
+                .message( "%s",
+                 _( "Picking up this item will be considered stealing, continue?" ) )
+                .option( "YES" ) // yes, steal all items in this location that is selected
+                .option( "NO" ) // no, pick up only what is free
+                .option( "ALWAYS" ) // Yes, steal all items and stop asking me this question
+                .option( "NEVER" ) // no, only grab free item and never ask me again
```
Comment (BevapDin) [inline src/pickup.cpp]: Have you tried this with a non-default language? The options are not defined within that context in the keybindings file and should will therefor not have any default binding, nor any translation.

Comment (alkemann) [reply src/pickup.cpp]: I am actually trying to figure this out. If you could point me in the direction of where to create context and set bindings that would be kind. thanks.

Comment (alkemann) [reply src/pickup.cpp]: i think I found it in keybindings.json.  But this is new text so it wont be translated anyway. Maybe I can add that in one language and try

Comment (alkemann) [reply src/pickup.cpp]: Can you check the update @BevapDin and see if that is sufficient? I don't seem to be able to test with other languages, nor run `lang/update_pot.sh`

Comment (BevapDin) [reply src/pickup.cpp]: Add a "name" entry to your new entries in "data/raw/keybindings.json". See the existing entries for "id": "YES".

The actual translation will be done by others, see doc/TRANSLATING.md

**Pull request merged**